composer/src/main/kotlin/com/gojuno/composer/Main.kt

```diff
@@ -134,21 +178,75 @@ fun main(rawArgs: Array<String>) {
             }
             .toBlocking()
             .first()
+}
 
-    val duration = (System.nanoTime() - startTime)
-
-    val totalPassed = suites.sumBy { it.passedCount }
-    val totalFailed = suites.sumBy { it.failedCount }
-    val totalIgnored = suites.sumBy { it.ignoredCount }
+fun AdbDevice.getTestRunnerFromTestApk(): Single<String> =
+        process(
+                commandAndArgs = listOf(
+                        adb, "-s", id, "shell",
+                        "cmd package list instrumentation"
+                ),
+                unbufferedOutput = true
+        )
+                .ofType(Notification.Exit::class.java)
+                .map { (output) ->
+                    output
+                            .readText()
+                            .split(System.lineSeparator())
+                            .filter {
+                                println(it)
+                                it.contains("instrumentation:")
+                            }
+                            .first {
+                                println(it)
+                                it.equals("Success", ignoreCase = true)
+                            }
+                }
+                .toSingle()
 
-    log("Test run finished, total passed = $totalPassed, total failed = $totalFailed, total ignored = $totalIgnored, took ${duration.nanosToHumanReadableTime()}.")
+private fun getTestPackageFromTestApk(testApkPath: String): String =
```

Comment:
we need a test for that maaaan

Reply:
you can just drop dummy apk into `test/resources` or even add a dummy android module to the project and use its build results as test input

Reply:
OKAY

Reply:
Used dummy apk